### PR TITLE
Finished porting of r_sub_thr & r_pub_thr

### PR DESCRIPTION
**Pull request by YuanYuYuan**
This PR consists of porting of r_sub_thr & r_pub_thr. 
- [x] Compilation passed.
- [x] `cargo fmt` & `cargo clippy` were checked. 
- [x] Benchmark with v5 is shown below.

throughput-comparison

### CODE REVIEW

**Review by eddie9712 — APPROVED** [review]
LGTM

File: throughput/src/bin/r_pub_thr.rs [hunk 1]
```diff
@@ -48,29 +60,34 @@ async fn main() {
...
-    config.insert(ZN_MULTICAST_SCOUTING_KEY, "false".to_string());
-    config.insert(ZN_PEER_KEY, opt.locator);
 
     let my_primitives = Arc::new(DummyPrimitives::new());
 
-    let runtime = Runtime::new(0u8, config, None).await.unwrap();
+    let runtime = Runtime::new(config).await.unwrap();
     let primitives = runtime.router.new_primitives(my_primitives);
 
+    let rid = KeyExpr::from(1);
```
Comment (william-wyliang) [inline throughput/src/bin/r_pub_thr.rs]: Please check if this line needs to be after the `decl_resource()`

**Review by william-wyliang — COMMENTED** [review]
One inline comment is added.

**Review by william-wyliang — APPROVED** [review]
LGTM

**Pull request merged**